src/common/CommonUtils.ts

```diff
@@ -78,5 +79,9 @@ export class CommonUtils {
         }
     }
 
+    public static castAsRequirement(input: any): Requirement {
```

Comment:
The caller should ensure that what they pass in can indeed be casted to `Requirement`

Reply:
What happens when `input` can't be casted to `Requirement`? This function seems to guarantee that the casting would succeed on all inputs.

Reply:
We're in charge of calling this and we would have to ensure it is called with correct input. But this whole thing is going away now that we're taking Kevin's approach